File: parcialito_4/recu_ej3.c
// 3. Implementar en C una función que reciba dos arreglos de enteros (y sus respectivos tamaños), 
// y que imprima los números del primer arreglo que **no** están presentes en el segundo.
// Ejemplo:
// int numeros[] = {1, 2, 7, 2, 3, 5};
// int ignorar[] = {2, 3};
// imprimir_faltantes(numeros, 6, ignorar, 2);
// // Salida:
// 1
// 7
// 5

// Escribir el codigo debajo de esta linea

#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

void imprimir_faltantes(int numeros[], int largo1, int ignorar[], int largo2) {

    for (int i = 0; i < largo1; i++) {

        int j = 0;

        while (numeros[i] != ignorar[j] && j < largo2) {

            j++;
        }

        if (j == largo2) {
            printf("%d\n", numeros[i]);
        
        }
    }
}

// main para pruebas
int main() {
    int numeros[] = {1, 2, 7, 2, 3, 5};
    int ignorar[] = {2, 3};
    imprimir_faltantes(numeros, 6, ignorar, 2);
    return 0;
}

File: parcialito_1/recu_ej3.py
# Escribir una función que reciba un número entero n, y devuelva una matriz triangular superior de dimensión n x n, en forma de lista de listas, 
# cuyos elementos son los números del 1 a K en orden (siendo K la cantidad total de números a colocar). 
# Por ejemplo, para n = 4, debe devolver la siguiente matriz:
# [[1, 2, 3, 4],
#  [0, 5, 6, 7],
#  [0, 0, 8, 9],
#  [0, 0, 0, 10]]

# Escribir el codigo debajo de esta linea

def triangular_superior(n):

    resultado = []
    relleno = 1
    
    for i in range(n):

        sublista = []

        for j in range(n):

            if i > j:
                sublista.append(0)
            
            else:
                sublista.append(relleno)

                relleno += 1
        
        resultado.append(sublista)
                
    
    return resultado

# Pruebas
resultado_esperado = [[1, 2, 3, 4],[0, 5, 6, 7],[0, 0, 8, 9],[0, 0, 0, 10]]
assert triangular_superior(4) == resultado_esperado